S3 documentation revamp (#717)
rudderlabs/rudderstack-docs: docs/destinations/storage-platforms/amazon-s3.mdx

---
title: "Amazon S3"
description: Step-by-step guide on sending your event data from RudderStack to Amazon S3.
---

[Amazon S3](https://aws.amazon.com/s3/) (Simple Storage Service) is a cloud-based object storage service that allows customers and businesses to store their data securely and at scale.

RudderStack lets you configure Amazon S3 as a destination where you can seamlessly store your event data.

<div class="infoBlock">
Find the open source transformer code for this destination in the <a href="https://github.com/rudderlabs/rudder-transformer/tree/master/v0/destinations/s3">GitHub repository</a>.
</div>

## Setting up Amazon S3

Follow these steps to set up your S3 bucket before adding it as a destination in RudderStack:

1. Login to your [Amazon AWS S3 console](https://aws.amazon.com/console/).
2. Create a new bucket. Alternatively, you can also choose an already-existing bucket.

## Permissions

> There are two ways to give RudderStack the relevant permissions for writing in your bucket. You can choose any one of them based on your internal security policies.

Reviewer: we are mentioning 3 options below, can modify this

> There are two ways to give RudderStack the relevant permissions for writing in your bucket. You can choose any one of them based on your internal security policies.

Author: ```suggestion
There are three ways to give RudderStack the relevant permissions for writing in your bucket. You can choose any one of them based on your internal security policies.
```

        "AWS": "arn:aws:iam::ACCOUNT_ID:user/USER_ARN"
      },
      "Action": ["s3:PutObject", "s3:PutObjectAcl"],
      "Resource": ["arn:aws:s3:::BUCKET_NAME/*"]
    }
  ]
}
```

3. Finally, add the programmatic access credentials for the above-created IAM user to the environment of your RudderStack setup, as shown:

```markup
RUDDER_AWS_S3_COPY_USER_ACCESS_KEY_ID=<access_key_id>
RUDDER_AWS_S3_COPY_USER_ACCESS_KEY=<secret_access_key>
```

### S3 permissions for warehouse destinations

> If you're using your S3 bucket as an **intermediary object storage** for the <Link to="/data-warehouse-integrations/">warehouse destination</Link>, then you will need the following bucket permissions:

Reviewer: ```suggestion
If you're using your S3 bucket as an **intermediary object storage** for a <Link to="/data-warehouse-integrations/">warehouse destination</Link>, you will need the additional bucket permissions as below:
```

Author: `PutObject` is a permission already included in the Permissions section above. Here we are listing **all** the permissions required to use S3 as an intermediary object storage for sending events to a WH destination.

```json
"Action": [
    "s3:GetObject",
    "s3:PutObject",
    "s3:PutObjectAcl",
    "s3:ListBucket"
]
```

## Configuring S3 destination in RudderStack

Follow these steps to set up S3 as a destination in RudderStack:

> 1. From your [RudderStack dashboard](https://app.rudderstack.com/), add a source. Then, from the list of destinations, select **Attribution**.

Reviewer: ```suggestion
1. From your [RudderStack dashboard](https://app.rudderstack.com/), add a source. Then, from the list of destinations, select **Amazon S3**.
```

2. Assign a name to the destination and click on **Continue**.

### Connection settings

In the **Connection Settings** page, enter the following settings to configure the S3 destination: 

<img src="../../assets/event-stream-destinations/s3-connection-settings.png" alt="S3 connection settings" />

- **S3 Bucket Name**: Enter your S3 bucket name.

> - **Prefix**: 

Reviewer 2: prefix content is empty
